localstack/services/awslambda/event_source_listeners/stream_event_source_listener.py

```diff
@@ -198,63 +200,75 @@ def _listen_to_shard_and_invoke_lambda(self, params: Dict):
             num_invocation_failures = 0
 
             while lock_discriminator in self._STREAM_LISTENER_THREADS:
-                records_response = stream_client.get_records(
-                    ShardIterator=shard_iterator, Limit=batch_size
-                )
-                records = records_response.get("Records")
-                event_filter_criterias = self._get_lambda_event_filters_for_arn(
-                    function_arn, stream_arn
-                )
-                if len(event_filter_criterias) > 0:
-                    records = filter_stream_records(records, event_filter_criterias)
-
-                should_get_next_batch = True
-                if records:
-                    payload = self._create_lambda_event_payload(
-                        stream_arn, records, shard_id=shard_id
-                    )
-                    is_invocation_successful, status_code = self._invoke_lambda(
-                        function_arn, payload, lock_discriminator, parallelization_factor
+                try:
+                    records_response = stream_client.get_records(
+                        ShardIterator=shard_iterator, Limit=batch_size
                     )
-                    if is_invocation_successful:
-                        should_get_next_batch = True
+                except ClientError as e:
+                    if "AccessDeniedException" in str(e):
+                        LOG.warning(
+                            "Insufficient permissions to get records from stream %s: %s",
```

Comment:
Are these types of IAM exceptions logged somewhere else on AWS?

Reply:
Might be in cloudtrail, but as it is a data operation, probably not even there. So no, it's very hard in AWS to detect if something like this happens.

This is why in AWS you have a test for the permissions before you create the event source mappings, but that is kind of lambda specific. We do not have that yet, but it is definitely planned to be added.